diff --git a/lib/recommend-version.js b/lib/recommend-version.js
index a982adfc92b52fe15987336390fa618afe83574e..90a19ead33f696421b818abbd7b33f31f4835888 100644
--- a/lib/recommend-version.js
+++ b/lib/recommend-version.js
@@ -82,6 +82,8 @@ function recommendVersion(pkg, type, { changelogPreset, rootPath, tagPrefix, pre
             //
             if (releaseType === "major") {
               releaseType = "minor";
+            } else if (releaseType === "minor") {
+              releaseType = "patch";
             }
           }
           log.verbose(type, "increment %s by %s", pkg.version, releaseType);
