// app/index.js
const callself = require('callself');
const interops = require('interops');
const moduleinterops = require('./moduleinterops');
const stableOrderJson = require('./assertion-generators/json.cjs');

// Generated by running v16.13.2: node assertion-generators/expected.cjs
// I chose this behavior because:
//  - it's more likely to occur in cjs codebase,
//  - it doesn't seem possible/reasonable to implement both behaviors depending on how the module is loaded,
const whatWouldNodejsRequireDo =
  '{"default":{"nameCollision":1},"nameCollision":3,"notDefault":2}';

// Would we rather reproduce the import behavior? see assertion-generators/expected.mjs
// const whatWouldNodejsImportDo = '{"default":{"default":{"nameCollision":1},"nameCollision":3,"notDefault":2},"nameCollision":3,"notDefault":2}'

function assertInteropNameCollisions({ moduleReference, expect, name }) {
  const whatWeGot = stableOrderJson(moduleReference);
  if (whatWeGot !== expect) {
    throw Error(`Exporting a field called 'default' from a cjs module '${name}' should not override all other fields.
    What we got     : ${whatWeGot}
    What we expected: ${expect}`);
  }
}

module.exports.assertions = {
  packageWithDefaultField() {
    assertInteropNameCollisions({
      name: 'interops',
      moduleReference: interops,
      expect: whatWouldNodejsRequireDo,
    });
  },
  moduleWithDefaultField() {
    assertInteropNameCollisions({
      name: 'moduleinterops',
      moduleReference: moduleinterops,
      expect: whatWouldNodejsRequireDo,
    });
  },
  packageReferencingItself() {
    try {
      callself.makeTheCall();
    } catch (e) {
      throw Error('Referencing exported functions should be possible', {
        cause: e,
      });
    }
  },
};
