// @noble/hashes/esm/sha2.js
/**
 * SHA2 hash function. A.k.a. sha256, sha384, sha512, sha512_224, sha512_256.
 * SHA256 is the fastest hash implementable in JS, even faster than Blake3.
 * Check out [RFC 4634](https://datatracker.ietf.org/doc/html/rfc4634) and
 * [FIPS 180-4](https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf).
 * @module
 */

export const sha256 = /* @__PURE__ */ ()=>'hello'
/** SHA2-224 hash function from RFC 4634 */
export const sha224 = /* @__PURE__ */ ()=>'hello'
/** SHA2-512 hash function from RFC 4634. */
export const sha512 = /* @__PURE__ */ ()=>'hello'
/** SHA2-384 hash function from RFC 4634. */
export const sha384 = /* @__PURE__ */ ()=>'hello'
/**
 * SHA2-512/256 "truncated" hash function, with improved resistance to length extension attacks.
 * See the paper on [truncated SHA512](https://eprint.iacr.org/2010/548.pdf).
 */
export const sha512_256 = /* @__PURE__ */ ()=>'hello'
/**
 * SHA2-512/224 "truncated" hash function, with improved resistance to length extension attacks.
 * See the paper on [truncated SHA512](https://eprint.iacr.org/2010/548.pdf).
 */
export const sha512_224 = /* @__PURE__ */ ()=>'hello'
